airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { useAsyncFn } from "react-use";
+import styled from "styled-components";
+
+import { Button } from "components/ui/Button";
+import { Switch } from "components/ui/Switch";
+
+import { useEnableConnection } from "hooks/services/useConnectionHook";
+
+interface IProps {
+  allowSync?: boolean;
+  enabled?: boolean;
+  isSyncing?: boolean;
+  isManual?: boolean;
+  id: string;
+  onSync: (id: string) => void;
+}
+
+const ProgressMessage = styled.div`
+  padding: 7px 0;
+`;
+
+export const StatusCellControl: React.FC<IProps> = ({ enabled, isManual, id, isSyncing, onSync, allowSync }) => {
```

Comment:
I would recommend moving this back to StatusCell and just adding a small wrapper for the div at the bottom of the file. Otherwise, this will cause a lot of conflicts with #19226, which will also add tests for the component